src/atmosphereFunctions.f90

```diff
@@ -72,11 +72,14 @@ pure function convertRHtoH2O( rh, temp, press ) result ( h2o )
     wvp = ( rh/100.0_DP ) * ( 6.116441_DP * 10.0_DP**((7.591386_DP * temp_c)/(temp_c + 240.7263_DP)) )
 
     ! Calculate volume of water vapour per volume of dry air using
-    ! Eq.18 (see Vaisala paper).
-    h2o_ppm = 1.0e+06_DP * wvp / (press - wvp)
-
-    ! convert ppm to molecule cm-3
-    h2o = h2o_ppm * calcAirDensity(press, temp) * 1.0e-06_DP
+    ! Eq.18 (see Vaisala paper). We don't use ppm here, because doing
+    ! so requires a multiplication by 1e6 then division by the same
+    ! in the final line. This incurs possible rounding. Instead, we
+    ! use 'parts per unit'
+    h2o_ppu = wvp / (press - wvp)
```

Comment:
@rs028 - I hope the reason for this change makes sense - having values of around 1e11, if we multiply by 1e6 and then by 1e-6, we lose precision as we can only hold about 16dp in `_DP` reals. I see this in practise on the "convertRHtoH2O: 50% relative humidity, 0C, pressure = A" unit test below, but making this change ensures we don't see such an effect. As `h2o_ppm` is not passed out of the function, it doesn't matter to the user.